pkg/git/git.go

```diff
@@ -0,0 +1,148 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/fileutil"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	IgnoreFile        = ".gitignore"
+	IgnoreDefaultMode = 0644
+)
+
+func git(dir string, args ...string) (string, error) {
+	cmd := exec.Command("git", args...)
+	cmd.Dir = dir
+	out, err := cmd.CombinedOutput()
+	return string(out), err
+}
+
+// IsRepository Return true if dir is a path to a directory in a git repository, false otherwise
+func IsRepository(dir string) bool {
+	_, err := git(dir, "rev-parse", "--is-inside-work-tree")
+	return err == nil
+}
+
+// GetRepositoryPath Returns the git repository root path if dir is a directory inside a git repository, otherwise returns error
+func GetRepositoryPath(dir string) (string, error) {
+	out, err := git(dir, "rev-parse", "--show-toplevel")
+	if err == nil {
+		return strings.TrimSpace(out), nil
+	}
+	if strings.Contains(out, "not a git repository") {
+		return "", ErrNotARepository
+	}
+	return "", fmt.Errorf("%s: %w", out, ErrGitError)
+}
+
+func createEntriesForIgnore(dir string, paths []string, exclude bool) ([]string, error) {
+	var entries []string
+	for _, p := range paths {
+		pathInRepo, err := filepath.Rel(dir, p)
+		if err != nil {
+			return nil, fmt.Errorf("%s :%w", p, err)
+		}
+		isDir, err := fileutil.IsDir(p)
+		if err != nil && !errors.Is(err, fs.ErrNotExist) {
+			return nil, fmt.Errorf("%s :%w", p, err)
+		}
+		if isDir {
+			pathInRepo = filepath.Join(pathInRepo, "*")
+		}
+		if exclude {
+			pathInRepo = "!" + pathInRepo
+		}
+		entries = append(entries, pathInRepo)
+	}
+	return entries, nil
+}
+
+func updateIgnoreFileSection(contents []byte, marker string, entries []string) []byte {
+	var newContent []byte
+	scanner := bufio.NewScanner(bytes.NewReader(contents))
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		newContent = append(newContent, []byte(fmt.Sprintln(line))...)
+		if line == marker {
+			for scanner.Scan() {
+				line = strings.TrimSpace(scanner.Text())
+				if line == "" {
+					break
+				}
+				if !slices.Contains(entries, line) {
+					newContent = append(newContent, []byte(fmt.Sprintln(line))...)
+				}
+			}
+			buffer := strings.Join(entries, fmt.Sprintln("")) + fmt.Sprintln("")
+			newContent = append(newContent, buffer...)
+		}
+	}
+
+	return newContent
+}
```

Comment:
Suggest the following to update/append the content:
```go
func updateIgnoreFileSection(contents []byte, marker string, entries []string) ([]byte, error) {
	// Read content and remove/collect entries under marker
	var (
		result   bytes.Buffer
		inMarker bool
		scanner  = bufio.NewScanner(bytes.NewReader(contents))
	)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		switch {
		case inMarker && line == "":
			inMarker = false
		case inMarker && !slices.Contains(entries, line):
			entries = append(entries, line)
		case !inMarker && line == marker:
			inMarker = true
		default:
			_, _ = fmt.Fprintln(&result, line)
		}
	}
	if err := scanner.Err(); err != nil {
		return nil, err
	}

	// Render marker and entries at the end
	_, _ = fmt.Fprintln(&result, marker)
	for _, entry := range entries {
		_, _ = fmt.Fprintln(&result, entry)
	}
	_, _ = fmt.Fprintln(&result)

	return result.Bytes(), nil
}
```

Reply:
Would prefer not to rewrite it to the end of the file for now. I will open another PR to also add an end marker